criblio/appscope
# Rename `filter` to `rules` #1520

### This PR

Replaces all uses of the word `filter` with the word `rules` where referring to the `filter file`.
Note: There were several uses of the word filter that needed to remain. I specifically ignored the term filter in the context of our config file where it referred to a regex filter or a config filter. This meant I had to take a slightly more manual approach.
Note to reviewer: I broke the changes down logically by commit to make this easier to review by looking at individual commits.

### Steps Followed

#### Step 1: Change all file and directory names
```
find . -name '*filter*'
```
(...then manually changed all applicable)


#### Step 2: Change all instances of 'filter' to 'rules' in files where appropriate

Commands examples:
```
make clean # remove all binary data and go modules	
grep -RIin filter # look for instances of filter that need to be replaced
sed -i 's/filter/rules/g' file # change one file
sed -i 's/Filter/Rules/g' file		
sed -i 's/FILTER/RULES/g' file	
find ./dir -type f -exec sed -i 's/filter/rules/g' {} + # change multiple files
find ./dir -type f -exec sed -i 's/Filter/Rules/g' {} +
find ./dir -type f -exec sed -i 's/FILTER/RULES/g' {} +
```

#### Changes:

No changes (no use of the word filter):
- / directory
- bin directory
- lib directory

No changes (use of the word filter OK):
- conf directory
- contrib directory
- docker directory
- docs directory
- os directory

Some changes (some use of the word filter OK):
- cli directory
- src
- test directory
- website directory

### QA Instructions
```
grep -RIin filter # check for remaining use of the word filter
grep -RIin rules # check for weirdness
git diff master # final confirmation
```

Closes #1520
